File: lib-database/src/main/java/com/ustadmobile/core/db/dao/DownloadJobItemParentChildJoinDao.java
package com.ustadmobile.core.db.dao;

import com.ustadmobile.lib.database.annotation.UmDao;
import com.ustadmobile.lib.database.annotation.UmInsert;
import com.ustadmobile.lib.database.annotation.UmQuery;
import com.ustadmobile.lib.db.entities.DownloadJobItemParentChildJoin;

import java.util.List;

@UmDao
public abstract class DownloadJobItemParentChildJoinDao {

    @UmQuery("SELECT DownloadJobItemParentChildJoin.* FROM DownloadJobItemParentChildJoin " +
            "WHERE djiChildDjiUid = :childDjiUid ")
    public abstract List<DownloadJobItemParentChildJoin> findParentsByChildUid(long childDjiUid);

    @UmQuery("SELECT DownloadJobItemParentChildJoin.* FROM DownloadJobItemParentChildJoin " +
            " LEFT JOIN DownloadJobItem ON DownloadJobItemParentChildJoin.djiParentDjiUid = DownloadJobItem.djiUid " +
            " WHERE DownloadJobItem.djiDjUid = :djUid")
    public abstract List<DownloadJobItemParentChildJoin> findParentChildJoinsByDownloadJobUids(int djUid);

    @UmInsert
    public abstract void insertList(List<DownloadJobItemParentChildJoin> joins);

}

File: lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/dao/DownloadJobItemParentChildJoinDao.kt
package com.ustadmobile.core.db.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.ustadmobile.lib.db.entities.DownloadJobItemParentChildJoin

@Dao
abstract class DownloadJobItemParentChildJoinDao {

    @Query("SELECT DownloadJobItemParentChildJoin.* FROM DownloadJobItemParentChildJoin " + "WHERE djiChildDjiUid = :childDjiUid ")
    abstract fun findParentsByChildUid(childDjiUid: Int): List<DownloadJobItemParentChildJoin>

    @Query("SELECT DownloadJobItemParentChildJoin.* FROM DownloadJobItemParentChildJoin " +
            " LEFT JOIN DownloadJobItem ON DownloadJobItemParentChildJoin.djiParentDjiUid = DownloadJobItem.djiUid " +
            " WHERE DownloadJobItem.djiDjUid = :djUid")
    abstract fun findParentChildJoinsByDownloadJobUids(djUid: Int): List<DownloadJobItemParentChildJoin>

    @Insert
    abstract fun insertList(joins: List<DownloadJobItemParentChildJoin>)

    @Insert
    abstract fun insert(jobRunItem: DownloadJobItemParentChildJoin): Long

}
